packages/redux-query-react/src/hooks/use-requests.js

```diff
@@ -0,0 +1,129 @@
+    .map(queryKey => queryConfigByQueryKey.get(queryKey))
+    .filter(Boolean);
+
+  return { cancelKeys, requestQueryConfigs };
+};
+
+const useRequests = (
+  providedQueryConfigs: ?Array<QueryConfig>,
+): [QueriesState, () => ?Promise<ActionPromiseValue>] => {
+  const reduxDispatch = useDispatch();
+
+  const previousQueryConfigs = React.useRef<Array<QueryConfig>>([]);
+
+  // This hook manually tracks the pending state, which is synchronized as precisely as possible
+  // with the Redux state. This may seem a little hacky, but we're trying to avoid any asynchronous
+  // synchronization. Hence why the pending state here is using a ref and it is updated via a
+  // synchronous callback, which is called from the redux-query query middleware.
+  const pendingRequests = React.useRef<Set<QueryKey>>(new Set());
+
+  // These "const callbacks" are memoized across re-renders. Unlike useCallback, useConstCallback
+  // guarantees memoization, which is relied upon elsewhere in this hook to explicitly control when
+  // certain side effects occur.
+  const dispatchRequestToRedux = useConstCallback((queryConfig: QueryConfig) => {
+    const promise = reduxDispatch(requestAsync(queryConfig));
+
+    if (promise) {
+      const queryKey = getQueryKey(queryConfig);
+
+      if (queryKey) {
+        pendingRequests.current.add(queryKey);
+      }
+    }
+  });
+
+  const dispatchCancelToRedux = useConstCallback((queryKey: QueryKey) => {
+    if (pendingRequests.current.has(queryKey)) {
+      reduxDispatch(cancelQuery(queryKey));
+      pendingRequests.current.delete(queryKey);
+    }
+  });
+
+  // Query configs are memoized based on query key. As long as the query keys in the list don't
+  // change, the query config list won't change.
+  const queryConfigs = useMemoizedQueryConfigs(providedQueryConfigs, (queryKey: QueryKey) => {
+    pendingRequests.current.delete(queryKey);
+  });
+
+  // This is an object containing two variables, isPending and isFinished, these apply to all queries.
+  // If any queries are pending, isPending is true, and
+  // unless all queries are finished, isFinished will be false.
+  const queriesState = useQueriesState(queryConfigs);
+
+  const forceRequest = React.useCallback(() => {
+    queryConfigs.forEach(requestReduxAction => {
+      dispatchRequestToRedux({
+        ...requestReduxAction,
+        force: true,
+      });
+    });
+  }, [dispatchRequestToRedux, queryConfigs]);
+
+  React.useEffect(() => {
+    // Whenever the list of query configs change, we need to manually diff the query configs
+    // against the previous list of query configs. Whatever was there and is no longer, will be
+    // cancelled. Whatever is new, will turn into a request.
+    const { cancelKeys, requestQueryConfigs } = diffQueryConfigs(
+      previousQueryConfigs.current,
+      queryConfigs,
+    );
+
+    requestQueryConfigs.forEach(dispatchRequestToRedux);
+    cancelKeys.forEach(queryKey => dispatchCancelToRedux(queryKey));
```

Comment:
I have a little concern here. Because I also wrote my own `useRequests` for my company. One edge case is if we first mount a component which has requests [A, B, C] and then we immediately unmount this component so that [A,B,C] will be cancelled (Assume [A,B,C] are all pending before unmounting). If you re-mount that component (for example, reopen that page), it will not request any [A,B,C] as they are already got cancelled. I am not sure whether this code will solve this problem

Reply:
Basically, I saw in `useRequest` we did some transform on queryConfig, 
```
{retry: true}
```
https://github.com/amplitude/redux-query/blob/master/packages/redux-query-react/src/hooks/use-request.js#L36-L44 

To solve this problem, maybe we also need to do this here?

Reply:
Ah, I am currently setting `{retry: true}` in the `use-memoized-query-configs.js`. So the retry logic is there. This should be moved out and passed into the memoize function. It still has the retry true, but the memoize should be agnostic to this, like how the current `use-request.js` works. I will change this.

Reply:
I think that's okay, as long as we add `{retry: true}`